File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/utility/ProcessResult.java
package org.nirvanasms.api.core.utility;

import org.nirvanasms.api.core.entity.StatusCode;

public class ProcessResult<T> {
	private boolean isSuccess;
	private String response;
	private String description;
	private T data;
	
	public ProcessResult()
    {

    }
	
	public ProcessResult(String response)
    {
        this.response = response;
        setDescriptionAndSuccess(response);
    }
	
	public ProcessResult(String response, T data)
    {
        this.response = response;
        setDescriptionAndSuccess(response);
        this.data = data;
    }

    void setDescriptionAndSuccess(String response)
    {
        String code = response.substring(0, 2);
        StatusCode status = StatusCodes.getByCode(code);
        this.description = String.format("İşlem Kodu: %s\nİşlem Açıklaması:%s", status.getCode(), status.getDescription());
        this.isSuccess = status.getIsSuccess();
    }

	public boolean isSuccess() {
		return isSuccess;
	}

	public void setSuccess(boolean isSuccess) {
		this.isSuccess = isSuccess;
	}

	public String getResponse() {
		return response;
	}

	public void setResponse(String response) {
		this.response = response;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public T getData() {
		return data;
	}

	public void setData(T data) {
		this.data = data;
	}
	
}

File: NirvanaSMS.Api.CSharp/Core/Utility/ProcessResult.cs
using NirvanaSMS.Api.Core.Entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NirvanaSMS.Api.Core.Utility
{
    public class ProcessResult<T>
    {
        public bool IsSuccess { get; set; }
        public string Response { get; set; }
        public string Description { get; set; }
        public T Data { get; set; }

        public ProcessResult()
        {

        }

        public ProcessResult(string response)
        {
            Response = response;
            setDescriptionAndSuccess(response);
        }

        public ProcessResult(string response, T data)
        {
            Response = response;
            setDescriptionAndSuccess(response);
            Data = data;
        }

        void setDescriptionAndSuccess(string response)
        {
            string code = response.Substring(0, 2);
            StatusCode status = StatusCodes.GetByCode(code);
            Description = string.Format("İşlem Kodu: {0}\nİşlem Açıklaması:{1}", status.Code, status.Description);
            IsSuccess = status.IsSuccess;
        }
    }

    public class ProcessResult
    {
        public bool IsSuccess { get; set; }
        public string Response { get; set; }
        public string Description { get; set; }

        public ProcessResult()
        {

        }

        public ProcessResult(string response)
        {
            Response = response;
            setDescriptionAndSuccess(response);
        }

        void setDescriptionAndSuccess(string response)
        {
            if (response.Length >= 2)
            {
                string code = response.Substring(0, 2);
                StatusCode status = StatusCodes.GetByCode(code);
                Description = string.Format("İşlem Kodu: {0}\nİşlem Açıklaması:{1}", status.Code, status.Description);
                IsSuccess = status.IsSuccess;
            }
            else
                IsSuccess = false;
        }
    }
}
